File: src/main/java/util/grid/Bounds.java
package util.grid;

import java.util.stream.Stream;

public record Bounds(int minX, int maxX, int minY, int maxY) {

    public boolean isInbound(Point point) {
        return this.minX <= point.x() && point.x() <= maxX &&
            this.minY <= point.y() && point.y() <= maxY;
    }

    public Stream<Point> filterInBounds(Stream<Point> points) {
        return points.filter(this::isInbound);
    }
}

File: util/src/main/java/util/grid/Bounds.kt
package util.grid

import java.util.stream.Stream

@JvmRecord
data class Bounds(val minX: Int, val maxX: Int, val minY: Int, val maxY: Int) {

    fun isInbound(point: Point): Boolean {
        return (minX..maxX).contains(point.x) && (minY..maxY).contains(point.y)
    }

    fun filterInBounds(points: Stream<Point>): Stream<Point> {
        return points.filter {
            this.isInbound(it)
        }
    }
}